### PR TITLE
Debug ability to bypass QA

### PR DESCRIPTION
**Pull request by timabell**
To make it easier to work on the apprenticeship features during local development.

### CODE REVIEW

**Review by gunndabad — COMMENTED** [review]
We tend to use 'CD_' prefix for environment variables.

Might be simpler to shim `ApprenticeshipQAStatusExtensions.ValueOrDefault` rather than this one query (to protect us if/when other queries also return this data).

File: src/Dfc.CourseDirectory.Core/DataStore/Sql/QueryHandlers/GetProviderApprenticeshipQAStatusHandler.cs [hunk 1]
```diff
@@ -19,6 +19,12 @@ public class GetProviderApprenticeshipQAStatusHandler :
 
             var paramz = new { query.ProviderId };
 
+#if DEBUG
+            if (System.Environment.GetEnvironmentVariable("BYPASS_QA")!=null && bool.Parse((System.Environment.GetEnvironmentVariable("BYPASS_QA"))))
```
Comment (gunndabad) [inline src/Dfc.CourseDirectory.Core/DataStore/Sql/QueryHandlers/GetProviderApprenticeshipQAStatusHandler.cs:23]:
```suggestion
            if (bool.TryParse(System.Environment.GetEnvironmentVariable("BYPASS_QA", out var bypassQA) && bypassQA)
```

Comment (timabell) [reply src/Dfc.CourseDirectory.Core/DataStore/Sql/QueryHandlers/GetProviderApprenticeshipQAStatusHandler.cs]: Cheers. Rebased and switched to new out-var syntax.

**Comment by timabell** [comment]
> We tend to use 'CD_' prefix for environment variables.

Sure; is there a reason for namespacing them that I'm unaware of?

> Might be simpler to shim ApprenticeshipQAStatusExtensions.ValueOrDefault rather than this one query (to protect us if/when other queries also return this data).

I tried modifying [ValueOrDefault](https://github.com/SkillsFundingAgency/dfc-coursedirectory/blob/a87afc044d9987fb57348b78ee191c44eae4c6dc/src/Dfc.CourseDirectory.Core/Models/ApprenticeshipQAStatus.cs#L42), I got the "passed" notification on the dashboard but not the links to the apprenticeships so I guess not everything is using that extension method currently.

**Comment by gunndabad** [comment]
> > We tend to use 'CD_' prefix for environment variables.
> 
> Sure; is there a reason for namespacing them that I'm unaware of?
> 
The .NET ones are prefixed so I followed the same convention. I assume it's so their purpose is obviously linked to a particular application but not sure.
> > Might be simpler to shim ApprenticeshipQAStatusExtensions.ValueOrDefault rather than this one query (to protect us if/when other queries also return this data).
> 
> I tried modifying [ValueOrDefault](https://github.com/SkillsFundingAgency/dfc-coursedirectory/blob/a87afc044d9987fb57348b78ee191c44eae4c6dc/src/Dfc.CourseDirectory.Core/Models/ApprenticeshipQAStatus.cs#L42), I got the "passed" notification on the dashboard but not the links to the apprenticeships so I guess not everything is using that extension method currently.
> 
Fair enough.
> Selection_20201123-01-c639cc639cc639c

**Review by gunndabad — APPROVED** [review]

**Pull request merged**